src/main/java/org/opentripplanner/middleware/OtpMiddlewareMain.java

```diff
@@ -131,9 +131,12 @@ private static void initializeHttpEndpoints() throws IOException, InterruptedExc
                 return "OK";
             });
 
-        // Security checks for admin and /secure/ endpoints.
+        // Security checks for admin and /secure/ endpoints. Excluding /authenticate so that API users can obtain a
+        // bearer token to authenticate against all other /secure/ endpoints.
         spark.before(API_PREFIX + "/secure/*", ((request, response) -> {
-            if (!request.requestMethod().equals("OPTIONS")) Auth0Connection.checkUser(request);
+            if (!request.requestMethod().equals("OPTIONS") && !request.pathInfo().endsWith("/authenticate")) {
```

Comment:
This should import and check the constant `AUTHENTICATE_PATH`. More precisely, it should probably check that the path is API_USER_PATH + AUTHENTICATE_PATH (or whatever the correct concatenation is).

Reply:
API_USER_PATH + AUTHENTICATE_PATH is correct. Updated.